### PR TITLE
[SPARK-11804][PYSPARK] Exception raise when using Jdbc predicates opt…

### PR DESCRIPTION
**Pull request by zjffdu**
…ion in PySpark

### CODE REVIEW

**Comment by rxin** [comment]
cc @davies

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 from pyspark.rdd import ignore_unicode_prefix
 from pyspark.sql.column import _to_seq
 from pyspark.sql.types import *
+import utils
```
Comment (davies) [inline python/pyspark/sql/readwriter.py]: from pyspark.sql.utils import toArray

File: python/pyspark/sql/utils.py [hunk 2]
```diff
@@ -71,3 +71,10 @@ def install_exception_handler():
     patched = capture_sql_exception(original)
     # only patch the one used in in py4j.java_gateway (call Java API)
     py4j.java_gateway.get_return_value = patched
+
+
+def toJArray(gateway, jtype, arr):
```
Comment (davies) [inline python/pyspark/sql/utils.py]: docstring?

**Comment by davies** [comment]
LGTM

**Pull request closed without merging**